### PR TITLE
build: remove myst-nb dependency

### PR DESCRIPTION
**Pull request by samjwu**
jupyter notebooks support not used or needed at this time

also allows myst-parser upgrade to version 1.0.0

### CODE REVIEW

**Review by saadrahim — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,4 @@ dist/**/*
 **/*.egg-info/**/*
 .sass-cache/*
 /build
+/_readthedocs/
```
Comment (Maetveis) [inline .gitignore]: Nope.

File: src/rocm_docs/core.py [hunk 2]
```diff
@@ -46,7 +46,7 @@ def update_body(body: Dict[str, Any]) -> None:
 
 @_config_updater
 def _ConfigExtend(key: str, app: Sphinx, default: T) -> None:
-    getattr(app.config, key).extend(default)
+    list(getattr(app.config, key)).extend(default)
```
Comment (Maetveis) [inline src/rocm_docs/core.py]: This won't have the desired effect, you're just creating a new list modifying it and throwing it away.

`myst_enable_extensions` is now a set, see [myst_parser configuration options](https://myst-parser.readthedocs.io/en/latest/configuration.html#global-configuration). You should change the value on line 79 to a set and introduce a helper for merging sets something like
```python
@_config_updater
def _ConfigUnion(key: str, app: Sphinx, default: Set[T]) -> None:
    gettattr(app.config, key).update(default)
```

**Review by Maetveis — CHANGES_REQUESTED** [review]

File: pyproject.toml [hunk 3]
```diff
@@ -24,14 +24,13 @@ dependencies = [
   "PyGithub>=1.58.1",
   "sphinx>=5.3.0",
   "breathe>=4.34.0",
-  "sphinx-design>=0.3.0",
+  "myst-parser>=1.0.0",
```
Comment (Maetveis) [inline pyproject.toml]:
```suggestion
  "myst-parser[linkify]>=1.0.0",
```
You dropped the "linkify" feature, this is causing the CI to fail.

File: src/rocm_docs/core.py (hunk 2 above)
Comment (samjwu) [reply src/rocm_docs/core.py]: Thanks for pointing that out.

File: .gitignore (hunk 1 above)
Comment (samjwu) [reply .gitignore]: Removed in https://github.com/RadeonOpenCompute/rocm-docs-core/pull/140/commits/9ad2c7f3d6cbfe368372416fa2d19b5c1644d8c9

File: pyproject.toml (hunk 3 above)
Comment (samjwu) [reply pyproject.toml]: Thanks. Fixed in [fc71a63](https://github.com/RadeonOpenCompute/rocm-docs-core/pull/140/commits/fc71a634547fdfdf89c9a0c81fc64c4671b45628)

**Pull request merged**